src/app/Routes.tsx

```diff
@@ -64,101 +65,103 @@ const Routes = (): JSX.Element => (
       <Route
         key={path}
         path={path}
-        render={() => (
+        element={
           <ErrorBoundary>
             <Controllers />
           </ErrorBoundary>
-        )}
+        }
       />
     ))}
-    {[devicesURLs.devices.index, devicesURLs.device.index(null, true)].map(
-      (path) => (
-        <Route
-          key={path}
-          path={path}
-          render={() => (
-            <ErrorBoundary>
-              <Devices />
-            </ErrorBoundary>
-          )}
-        />
-      )
-    )}
+    <Route
+      path={devicesURLs.devices.index}
+      element={
+        <ErrorBoundary>
+          <DeviceList />
+        </ErrorBoundary>
+      }
+    />
+    <Route
+      path={`${devicesURLs.device.index(null, true)}/*`}
+      element={
+        <ErrorBoundary>
+          <DeviceDetails />
```

Comment:
These components contain often shared state between the child routes and with this approach (unlike this one: https://github.com/canonical-web-and-design/maas-ui/pull/3957) we can keep this structure.